### PR TITLE
feat: intrinsic_evaluation_mode_visual_indicators

### PR DESCRIPTION
**Pull request by SergioReyesSan**
## Description

The proposed features in this PR are:

Add some UI elements to the main window, indicating useful information like: speed, angles, and aspect ratio of the board, as well as indicators for the samples added according to the board size and skew angle.

Fix bugs when reading intrinsic parameter files.

Add a linearity heatmap to check the image distortion using the calibration board.
 
Save the training and evaluation points used for calibration and the parameters used in a file.

Loading the intrinsics file enables the evaluation mode, which, by default, modifies the elements shown on the main UI.

## Related links

No related links for now.

## Tests performed

Tested with rosbags that contain chessboard and dot calibration boards, however, some parameters need to be adjusted like the percentage for color transition in skew and size indicators for real-time detections.

## Notes for reviewers

Some parameters need to be adjusted like the color transition for skew and size indicators for real-time detections.

Angle filtering and some extra filtering might be needed.

I have tried to delete spaces on empty lines however I might skipped some of them.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by knzo25** [comment]
@SergioReyesSan 
After amadeusz's PR some conflicts appeared in your PR. Can you check those?
(those conflicts are a bit weird, since I made a branch starting in yours and merging tier4/universe and there were no conflicts. If you can not check the conflicts by tomorrow, we can use the branched I prepared for the experiment)

**Comment by knzo25** [comment]
From now now, please follow conventional commits as well
https://www.conventionalcommits.org/en/v1.0.0/

Also, please sign you commits with `git commit -s`

File: calibrators/intrinsic_camera_calibrator/intrinsic_camera_calibrator/intrinsic_camera_calibrator/views/initialization_view.py [hunk 1]
```diff
@@ -136,6 +136,9 @@ def load_intrinsics_button_callback():
             )
             self.initial_intrinsics = load_intrinsics(intrinsics_path)
             self.evaluation_radio_button.setEnabled(True)
+            # self.training_radio_button.setChecked(False)
```
Comment (knzo25) [inline calibrators/intrinsic_camera_calibrator/intrinsic_camera_calibrator/intrinsic_camera_calibrator/views/initialization_view.py:139]: Can you delete this commented line?

**Comment by knzo25** [comment]
note: for this PR or the next, the distortion model and the number of coefficients is not being handled correctly in I/O

File: calibrators/intrinsic_camera_calibrator/intrinsic_camera_calibrator/intrinsic_camera_calibrator/camera_calibrator.py [hunk 2]
```diff
@@ -787,6 +852,22 @@ def process_evaluation_results(
...
 
+    def save_parameters(self, filename):
+        data_coll_params = self.data_collector.parameters_value()
+        board_params = self.board_parameters.parameters_value()
+        detector_params = self.detector.parameters_value()
+        calibrator_type = self.calibrator_type_combobox.currentData()
+        print(calibrator_type.value["name"], flush=True)
+        calib_params = self.calibrator_dict[calibrator_type].parameters_value()
+        with open(filename, "w") as file:
+            yaml.dump({"board_parameters": board_params}, file, default_flow_style=False)
```
Comment (knzo25) [inline calibrators/intrinsic_camera_calibrator/intrinsic_camera_calibrator/intrinsic_camera_calibrator/camera_calibrator.py:863]: I think it is better to perform a single dump of a dictionary containing all the elements

**Comment by knzo25** [comment]
@SergioReyesSan 
If your new PR replaces this one, please close and reference the new one.

**Comment by knzo25** [comment]
@SergioReyesSan

**Comment by SergioReyesSan** [comment]
Replaced by the new PR
